src/coap_session.c

```diff
@@ -698,13 +705,14 @@ coap_session_create_client(
       &session->addr_info.local, &session->addr_info.remote)) {
       goto error;
     }
-  } else if (!COAP_DISABLE_TCP &&
```

Comment:
The reason we did the exclusion this way was better readability and less cluttering with pre-processor directives. The optimizer then should remove the dead code.

Reply:
I agree about the readability, however, the Window C compilers complained that `coap_socket_connect_tcp1() `was not there during linking so the optimizer was failing to get rid of the dead code (hence how I found that this fix is needed for windows with TCP code disabled).